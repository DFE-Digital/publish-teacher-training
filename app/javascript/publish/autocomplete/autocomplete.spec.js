/**
 * @jest-environment jsdom
 */

import initAutocomplete from './autocomplete'

describe('Autocomplete', () => {
  describe('initAutocomplete', () => {
    beforeEach(() => {
      document.body.innerHTML = `
         <div id="outer-container">
           <input type="text" id="input">
           <div id="autocomplete-container"></div>
         </div>
       `
      const schoolTemplate = (result) => result && `${result.name}`

      initAutocomplete(
        'autocomplete-container',
        'input',

        { path: '/endpoint', template: { inputValue: schoolTemplate, suggestion: schoolTemplate } }
      )
    })

    it('should instantiate an autocomplete', () => {
      expect(document.querySelector('#outer-container')).toMatchSnapshot()
    })
  })
})
