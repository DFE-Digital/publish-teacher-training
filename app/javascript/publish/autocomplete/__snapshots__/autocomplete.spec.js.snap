// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Autocomplete initAutocomplete should instantiate an autocomplete 1`] = `
<div
  id="outer-container"
>
  
           
  <input
    id="input"
    type="text"
  />
  
           
  <div
    id="autocomplete-container"
  />
  
         
</div>
`;
